Does not build with Boost 1.77

Build testing the OpenBSD ports tree with Boost 1.77 beta1. widelands 1.0 does not build with Boost 1.77 beta1 like so..

```
/pobj/widelands-1.0/bin/c++ -DUSE_XDG -DWL_USE_GLVND -I/pobj/widelands-1.0/widelands-1.0/src -Isrc -isystem /usr/X11R6/include -isystem /usr/local/include -isystem /usr/local/include/SDL2 -O2 -pipe -DNDEBUG  -std=c++11  -DNDEBUG -DNOPARACHUTE -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-conversion -Wno-exit-time-destructors -Wno-global-constructors -Wno-padded -Wno-sign-conversion -Wno-missing-noreturn -Wno-switch-enum -Wno-weak-vtables -Wno-unreachable-code -Wno-documentation -Werror=non-pod-varargs -Qunused-arguments -Wint-to-void-pointer-cast -Winconsistent-missing-override -Werror=format-security -Werror=return-type -Werror=uninitialized -MD -MT src/network/CMakeFiles/network.dir/bufferedconnection.cc.o -MF src/network/CMakeFiles/network.dir/bufferedconnection.cc.o.d -o src/network/CMakeFiles/network.dir/bufferedconnection.cc.o -c /pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.cc
In file included from /pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.cc:1:
/pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.h:313:7: error: no template named 'map' in namespace 'std'; did you mean 'max'?
        std::map<uint8_t, std::queue<std::vector<uint8_t>>> buffers_to_send_;
        ~~~~~^~~
             max
/usr/include/c++/v1/algorithm:2624:1: note: 'max' declared here
max(const _Tp& __a, const _Tp& __b, _Compare __comp)
^
In file included from /pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.cc:1:
/pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.h:313:7: error: non-friend class member 'max' cannot have a qualified name
        std::map<uint8_t, std::queue<std::vector<uint8_t>>> buffers_to_send_;
        ~~~~~^
/pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.h:313:7: error: C++ requires a type specifier for all declarations
/pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.h:313:53: error: expected ';' at end of declaration list
        std::map<uint8_t, std::queue<std::vector<uint8_t>>> buffers_to_send_;
                                                           ^
                                                           ;
/pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.h:221:3: error: use of undeclared identifier 'buffers_to_send_'
                buffers_to_send_[priority].push(v);
                ^
/pobj/widelands-1.0/widelands-1.0/src/network/bufferedconnection.cc:213:21: error: use of undeclared identifier 'buffers_to_send_'
        for (auto& entry : buffers_to_send_) {
                           ^
6 errors generated.
```

--- a/src/network/bufferedconnection.h
+++ b/src/network/bufferedconnection.h
@@ -20,6 +20,7 @@
 #ifndef WL_NETWORK_BUFFEREDCONNECTION_H
 #define WL_NETWORK_BUFFEREDCONNECTION_H
 
+#include <map>
 #include <memory>
 #include <mutex>
 #include <thread>

